x-pack/metricbeat/module/aws/rds/rds.go

```diff
@@ -172,6 +175,27 @@ func getDBInstancesPerRegion(svc rdsiface.ClientAPI) ([]string, map[string]DBDet
 			dbDetails.dbAvailabilityZone = *dbInstance.AvailabilityZone
 		}
 
+		// Get tags for each RDS instance
+		listTagsInput := rds.ListTagsForResourceInput{
+			ResourceName: dbInstance.DBInstanceArn,
+		}
+		reqListTags := svc.ListTagsForResourceRequest(&listTagsInput)
+		outputListTags, err := reqListTags.Send(context.TODO())
+		if err != nil {
+			m.Logger().Warn("ListTagsForResourceRequest failed, rds:ListTagsForResource permission is required for getting tags.")
+			dbDetailsMap[*dbInstance.DBInstanceIdentifier] = dbDetails
+			return dbInstanceIDs, dbDetailsMap, nil
+		}
+
+		for _, tag := range outputListTags.TagList {
+			// By default, replace dot "." using under bar "_" for tag keys and values
```

Comment:
I would add a check on tag here to check if it is nil before letting it to be dereferenced later on. sth like:
```
if tag == nil {
  continue
}
```

Reply:
I tested with a rds with no tag. Seems like when `outputListTags.TagList` is empty, for loop will not be continued here. So we are safe without checking `tag==nil` 😬